scripts/zones/Port_Jeuno/npcs/Squintrox_Dryeyes.lua

```diff
@@ -3,117 +3,534 @@
+            if kit == 0 then
+                kit = xi.item.ENFEEBLEMENT_KIT_OF_POISON + math.random(0,3)
+                player:setCharVar('ASA_kit', kit)
+            end
+            player:updateEvent(kit)
+        elseif option == 104 then -- Bird Key
+            local completedSeals =
+            (player:hasKeyItem(xi.ki.AMBER_COUNTERSEAL)    and 1 or 0) +
+            (player:hasKeyItem(xi.ki.AZURE_COUNTERSEAL)    and 1 or 0) +
+            (player:hasKeyItem(xi.ki.CERULEAN_COUNTERSEAL) and 1 or 0) +
+            (player:hasKeyItem(xi.ki.EMERALD_COUNTERSEAL)  and 1 or 0) +
+            (player:hasKeyItem(xi.ki.SCARLET_COUNTERSEAL)  and 1 or 0) +
+            (player:hasKeyItem(xi.ki.VIOLET_COUNTERSEAL)   and 1 or 0)
+
+            if completedSeals >= 3 then
+                player:setLocalVar('ASA_Status', completedSeals)
+                player:updateEvent(0, completedSeals, 2)
+            else
+                player:updateEvent(0, 0, 1)
+            end
+        elseif option == 105 then -- Bomb Key
+            local completedFragments =
+                (player:hasKeyItem(xi.ki.LUMINOUS_PURPLE_FRAGMENT) and 1 or 0) +
+                (player:hasKeyItem(xi.ki.LUMINOUS_YELLOW_FRAGMENT) and 1 or 0) +
+                (player:hasKeyItem(xi.ki.LUMINOUS_BLUE_FRAGMENT)   and 1 or 0) +
+                (player:hasKeyItem(xi.ki.LUMINOUS_BEIGE_FRAGMENT)  and 1 or 0) +
+                (player:hasKeyItem(xi.ki.LUMINOUS_RED_FRAGMENT)    and 1 or 0) +
+                (player:hasKeyItem(xi.ki.LUMINOUS_GREEN_FRAGMENT)  and 1 or 0)
+
+            if completedFragments ~= 6 then
+                player:updateEvent(0, 0, 0, 0, 0, 0, 3)
+            end
+        elseif option == 203 then -- 2nd page
+            local finishedACP = player:getCurrentMission(xi.mission.log_id.ACP) == xi.mission.id.acp.A_CRYSTALLINE_PROPHECY_FIN
+            local finishedAMK = player:getCurrentMission(xi.mission.log_id.AMK) == xi.mission.id.amk.A_MOOGLE_KUPO_DETAT_FIN
+            local finishedASA = player:getCurrentMission(xi.mission.log_id.ASA) == xi.mission.id.asa.A_SHANTOTTO_ASCENSION_FIN
+
+            -- Reminder that "True" means the option should be excluded from the player's menu
+            local arg1 = (not finishedACP                                                or
+                             (player:hasKeyItem(xi.ki.SEEDSPALL_ROSEUM)                 and
+                             player:hasKeyItem(xi.ki.SEEDSPALL_CAERULUM)                and
+                             player:hasKeyItem(xi.ki.SEEDSPALL_VIRIDIS))                and
+                                 2 or 0) +
+                         (not finishedACP                                                or
+                             (player:hasKeyItem(xi.ki.MARK_OF_SEED))                    and
+                                 4 or 0) +
+                         (not finishedACP                                                or
```

Comment:
I'd look into the pattern I mentioned before about this being a data table that's then iterated over. Remember you can nest tables, so you can have a table of KI requirements as an entry in the main table

Reply:
I think that is the route I will take. The same data is also used in the next event function in the same order so it should extend nicely